core-plugins/src/main/java/co/cask/hydrator/plugin/batch/action/OracleExportAction.java

```diff
@@ -0,0 +1,251 @@
+    scriptContent.append(validateQuery(config.queryToExecute) + "\n");
+    scriptContent.append("spool off" + "\n");
+    scriptContent.append("exit");
+    String tmpSqlScriptFile = config.tmpSQLScriptDirectory + "/tmpHydrator.sql";
+    //Set of commands to be executed in a session
+    String setPath = "export ORACLE_HOME=" + config.oracleHome + "; export ORACLE_SID=" + config.oracleSID + ";";
+    String tmpScriptCreationCommand = "echo '" + scriptContent.toString() + "' > " + tmpSqlScriptFile + ";";
+    String sqlPlusSpoolExecutionCommand = config.oracleHome + "/bin" + "/sqlplus -s " + dbConnectionString +
+      " @" + tmpSqlScriptFile + " |  sed 's/\\s*" + colSeparator + "\\s*/" + colSeparator + "/g';";
+    String tmpSpoolScriptRemovalCommand = "rm " + tmpSqlScriptFile + "";
+    return setPath + tmpScriptCreationCommand + sqlPlusSpoolExecutionCommand + tmpSpoolScriptRemovalCommand;
+  }
+
+  private String getColSeparator(String format) {
+    String columnSeparator = "";
+    switch (SeparatorFormat.valueOf(format)) {
+      case csv:
+        columnSeparator = ",";
+        break;
+      case tsv:
+        columnSeparator = "  ";
+        break;
+      case psv:
+        columnSeparator = "|";
+        break;
+    }
+    return columnSeparator;
+  }
+
+  /**
+   * Config class that contains all properties necessary to execute the SQLPLUs spool command.
+   */
+  public static class OracleExportActionConfig extends PluginConfig {
+
+    @Description("Host name of the remote DB machine")
+    @Macro
+    private String oracleServerHostname;
+
+    @Nullable
+    @Description("Port to use to SSH to the remote Oracle Host. Defaults to 22.")
+    @Macro
+    private Integer oracleServerPort;
+
+    @Description("Username to use to connect to the remote Oracle Host via SSH.")
+    @Macro
+    private String oracleServerUsername;
+
+    @Description("Password to use to connect to the remote Oracle Host via SSH.")
+    @Macro
+    private String oracleServerPassword;
+
+    @Description("Username to connect to the Oracle database.")
+    @Macro
+    private String dbUsername;
+
+    @Description("Password to connect the Oracle database.B")
+    @Macro
+    private String dbPassword;
+
+    @Description("Absolute path of the ``ORACLE_HOME`` environment variable on the Oracle server host." +
+      "This will be used to run the Oracle Spool utility.")
+    @Macro
+    private String oracleHome;
+
+    @Description("Oracle System ID(SID). This is used to uniquely identify a particular database on the system.")
+    @Macro
+    private String oracleSID;
+
+    @Description("Query to be executed for export." +
+      "For example: select * from test where name='cask';")
+    @Macro
+    private String queryToExecute;
+
+    @Description("Path to the directory where temporary SQL script needs to be created. It will be removed " +
+      "once the SQL command is executed.")
+    @Macro
+    private String tmpSQLScriptDirectory;
```

Comment:
should add a default for this. Perhaps default to ``/tmp`` and mention it in the description.